### PR TITLE
sockets: tls: Fix Kconfig mbedTLS dependencies

### LINKED ISSUE #21768: Make [CONFIG_NET_SOCKETS_SOCKOPT_TLS] dependent on [CONFIG_MBEDTLS] in menuconfig
**Describe the bug**
When you define CONFIG_NET_SOCKETS_SOCKOPT_TLS without CONFIG_MBEDTLS, zephyr has build errors in socket_tls.c.
Workaround is knowing the dependencies and add it to configuration

**To Reproduce**
1. build the dhcpv4_client sample application for the stm32f746g_disco
2. add to prj.conf

# Enable Sockets support
CONFIG_NET_SOCKETS=y
CONFIG_NET_SOCKETS_SOCKOPT_TLS=y

3. build will fail

**Expected behavior**
At configuration time before building, error should indicate configuration of socket depends on mbedtls and should not start building the source code

**Impact**
annoying and configuration not consistent

**Environment (please complete the following information):**
 - MacOS
 - using zephyr-v2.1.0

### PR DESCRIPTION
**Pull request by rlubos**
Sort out mbedTLS dependencies in sockets Kconfig. mbedTLS will
now be enabled when TLS sockets are enabled and native network
stack is enabled.

Fixes #21768

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

### CODE REVIEW

**Comment by rlubos** [comment]
I know there was a similar PR already opened, but it seemed to stall and I need this issue to be fixed sooner than later.

File: subsys/net/lib/sockets/Kconfig
Comment (ulfalizer) [inline subsys/net/lib/sockets/Kconfig]: Does `NET_SOCKETS_SOCKOPT` need `TLS_CREDENTIALS` and `MBEDTLS` to be enable if `NET_NATIVE` is enabled, or are they still optional?

If it needs them to be enabled, you could use `select`. Otherwise, `imply` is fine (which just turns them on by default).

File: subsys/net/lib/sockets/Kconfig
Comment (ulfalizer) [inline subsys/net/lib/sockets/Kconfig]: Ditto here.

**Comment by ulfalizer** [comment]
Could merge if `imply` makes sense (i.e., if it makes sense to turn the symbols off when `NET_NATIVE` is on) without waiting for approval. Could merge with `select` if that makes more sense too.

File: subsys/net/lib/sockets/Kconfig
Comment (rlubos) [reply subsys/net/lib/sockets/Kconfig]: They are required, at least for now there is no other implementaion for TLS sockets. Will switch to `select`.

**Review by ulfalizer — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**
